spec/label_spec.rb

```diff
@@ -0,0 +1,89 @@
+require_relative '../classes/Book/label'
+require_relative '../classes/item'
+
+describe Label do
+  describe 'unit tests for \'Label\' class' do
+    before(:each) do
+      # set the params to create a Label instance
+      @label_params_req = {
+        name: 'test label',
+        title: 'book label',
+        color: 'red'
+      }
+      @label_params_id = {
+        id: 1,
+        name: 'test label'
+      }
+      @new_label_req = Label.new(@label_params_req)
+      @new_label_id = Label.new(@label_params_id)
+
+      @label_instance = instance_double(Label)
+
+      # create an Item instance
+      item_params_id = {
+        id: 1,
+        publish_date: '2023-08-01'
+      }
+      @new_item = Item.new(item_params_id)
+    end
+
+    context 'when creating a new \'Label\'' do
+      it 'should create an instance of \'Label\' class' do
+        expect(@new_label_req).to be_an_instance_of Label
+        expect(@new_label_id).to be_an_instance_of Label
+      end
+
+      it 'should have a valid \'id\' not \'nil\' but an \'Integer\'' do
+        expect(@new_label_req.id).not_to be_nil
+        expect(@new_label_id.id).not_to be_nil
+        expect(@new_label_req.id).to be_an_instance_of(Integer)
+        expect(@new_label_id.id).to be_an_instance_of(Integer)
+      end
+
+      it 'should have a valid \'title\' not \'nil\' but a \'String\'' do
+        expect(@new_label_req.title).not_to be_nil
+        expect(@new_label_req.color).not_to be_nil
+        expect(@new_label_req.title).to be_an_instance_of(String)
+        expect(@new_label_req.title).to be_an_instance_of(String)
+        expect(@new_label_req.color).to be_an_instance_of(String)
+        expect(@new_label_req.color).to be_an_instance_of(String)
+      end
```

Comment:
- For this test, we are checking the value of `title`, don't mix it with the `color`
- Also they were using the same instance `@new_label_req` and you were using `new_label_id`
```suggestion
      it 'should have a valid \'title\' not \'nil\' but a \'String\'' do
        expect(@new_label_req.title).not_to be_nil
        expect(@new_label_id.title).not_to be_nil
        expect(@new_label_req.title).to be_an_instance_of(String)
        expect(@new_label_id.title).to be_an_instance_of(String)
      end
```

- It is necessary to create a new `it` block for testing the `color` as you did with `title`